Remove legacy bike access mapping

leonardehrenfried opened the pull request:
### Summary

This removes the parsing of legacy `bikes_allowed` information. The fields `trip_bikes_allowed` and `route_bikes_allowed` were discussed early in the GTFS spec process but have never made it into the specification.

It has been deprecated in onebusaway-gtfs for many years and removed in the latest version.

### Issue

None.

### Unit tests

Existing ones updated.

flaktack commented on application/src/main/java/org/opentripplanner/gtfs/mapping/BikeAccessMapper.java:
```diff
@@ -5,64 +5,23 @@
 
   public static BikeAccess mapForRoute(Route rhs) {
-    //noinspection deprecation
-    return mapValues(rhs.getBikesAllowed(), rhs.getRouteBikesAllowed());
+    return mapValues(rhs.getBikesAllowed());
   }
```
`bikes_allowed` only seems to be present in `trips.txt`. Should the route based handling also be removed?
https://gtfs.org/documentation/schedule/reference/#routestxt

leonardehrenfried replied:
I would also have a knock-on effect on the `Route` type so that would also require more discussion.

optionsome replied:
Pretty sure we don't have the bikes_allowed information in routes.txt in any of our data sets.

flaktack reviewed (commented):
The simple test GTFS contains both `route_bikes_allowed` and `trip_bikes_allowed` columns. Should those be removed, replaced with the standard versions?

https://github.com/opentripplanner/OpenTripPlanner/blob/5b5d92fe7e81c2fab07f2ff6ae8f7488427be71c/application/src/test/resources/gtfs/simple/routes.txt#L1

https://github.com/opentripplanner/OpenTripPlanner/blob/5b5d92fe7e81c2fab07f2ff6ae8f7488427be71c/application/src/test/resources/gtfs/simple/trips.txt#L1

leonardehrenfried commented:
> The simple test GTFS contains both `route_bikes_allowed` and `trip_bikes_allowed` columns. Should those be removed, replaced with the standard versions?
> 
> https://github.com/opentripplanner/OpenTripPlanner/blob/5b5d92fe7e81c2fab07f2ff6ae8f7488427be71c/application/src/test/resources/gtfs/simple/routes.txt#L1
> 
> https://github.com/opentripplanner/OpenTripPlanner/blob/5b5d92fe7e81c2fab07f2ff6ae8f7488427be71c/application/src/test/resources/gtfs/simple/trips.txt#L1

Good catch. Updated.
